File: Python/Algorithms/504.py
class Solution:
    def convertToBase7(self, num: int) -> str:
        if num == 0:
            return "0"
        num, neg = (num, False) if num > 0 else (-num, True)
        res = []
        while num:
            num, r = divmod(num, 7)
            res.append(r)
        if neg:
            res.append("-")
        return "".join(map(str, reversed(res)))

File: Rust/Algorithms/504.rs
impl Solution {
    pub fn convert_to_base7(num: i32) -> String {
        if num == 0 {
            return String::from("0");
        }
        
        let neg = if num > 0 { false } else { true };
        let mut num = if num > 0 { num } else { -num };
        
        let mut result = vec![];
        while num > 0 {
            result.push((num % 7).to_string());
            num /= 7;
        }
        if neg {
            result.push("-".to_string());
        }
        result.into_iter().rev().collect::<Vec<_>>().join("")
    }
}